### ISSUE TITLE
Specifying and installing extras

### ISSUE THREAD

**Issue by ewianda** [dillon-giacoppo/rules_python_external#19]
# Description 
Installing extras 

### Steps taken
generate requirements.txt 

    pip-compile --generate-hashes    requirements.in 
    # requirements.in
    apache_beam[gcp]

### What was expected
apache-beam should be installed with extras for gcp

### What happened:

```
' failed; build aborted: no such package '@my_deps//pypi__apache_beam[gcp]':
 BUILD file not found in directory 'pypi__apache_beam[gcp]' 
of external repository @my_deps. Add a BUILD file to a directory to mark it as a package.

```

**Comment by dillon-giacoppo** [comment]
There are a few things happening here. The first is that the bazel label should be specified _without_ the extras. In your case it would be:

```
deps = [requirement("apache_beam")] ## equivalent to deps = ["pypi__apache_beam"]
```

There is, however, an underlying bug. We pass the requirements.txt file directly into `pip wheel` and then create BUILD files for all the output `*.whl`. This approach is convenient as wheels have an easy to parse metadata format.

Unfortunately, we cannot determine the extras a user has requested without parsing the `requirements.txt` file directly. There are a lot of edge cases when parsing this file which makes it quite difficult and I was unable to find a good library.

The end result is we install the right dependencies, but we don't specify them transitively.

A solution to this is that we do naive parsing of the file and try a best-effort approach. A more comprehensive idea is to do a custom metadata format that makes the information easy to fetch. This is what is suggested by https://github.com/dillon-giacoppo/rules_python_external/issues/15. 

A workaround that will get it up and running right now is to use `all_requirements`:
 ```
load("@pip//:requirements.bzl", "all_requirements")

py_binary(
    name = "main",
    srcs = ["main.py"],
    deps = all_requirements +  [
        # Other deps here
    ],
)
```
This lists everything specified in the requirements.txt as a dependency.

**Comment by thundergolfer** [comment]
@dillon-giacoppo has this been fixed by #28?

**Comment by dillon-giacoppo** [comment]
Yep, just cut the release for it. Fixed in v0.1.4
